.github/workflows/cd.yml

```diff
@@ -0,0 +1,175 @@
+name: cd
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploy_infra:
+        description: 'Whether to deploy infrastructure'
+        default: true
+        required: true
+        type: boolean
+      deploy_app:
+        description: 'Whether to deploy app'
+        default: true
+        required: true
+        type: boolean
+  workflow_call:
+    inputs:
+      deploy_infra:
+        required: true
+        type: boolean
+      deploy_app:
+        required: true
+        type: boolean
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+
+concurrency: cd
+
+env:
+  IMAGE_NAME: 'keyserver'
+
+jobs:
+  deploy-infra-staging:
+    if: ${{ inputs.deploy_infra }}
+    runs-on: ubuntu-latest
+    environment:
+      name: infra/staging
+      url: https://staging.chat-keys.walletconnect.com/health
```

Comment:
We should extract `https://staging.chat-keys.walletconnect.com` to some variable

Reply:
True. Will do so as a follow up!